src/main/java/yoyo/MainWindow.java

```diff
@@ -0,0 +1,91 @@
+package yoyo;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+import java.util.Objects;
+
+/**
+ * Controller for the main GUI.
+ */
+public class MainWindow extends AnchorPane {
+
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
+
+    private YoyoAdapter yoyo;
+
+    private Image userImage;
+    private Image dukeImage;
+
+    @FXML
+    public void initialize() {
+        // Load images
+        try {
+            userImage = new Image(Objects.requireNonNull(this.getClass().getResourceAsStream("/images/DaUser.png")));
+            dukeImage = new Image(Objects.requireNonNull(this.getClass().getResourceAsStream("/images/DaDuke.png")));
+        } catch (Exception e) {
+            // Use null images - DialogBox should handle this gracefully
+            userImage = null;
+            dukeImage = null;
+        }
+
+        // Auto-scroll to bottom when content grows
+        dialogContainer.heightProperty().addListener((obs, oldVal, newVal) -> {
+            scrollPane.setVvalue(1.0);
+        });
+
+        // Add initial greeting
+        addBotLine("Hello! I'm Yoyo. How can I help?");
+    }
+
+    /**
+     * Injects the YoyoAdapter instance
+     */
+    public void setYoyoAdapter(YoyoAdapter y) {
+        yoyo = y;
+    }
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing
+     * Yoyo's reply and then appends them to the dialog container. Clears the
+     * user input after processing.
+     */
+    @FXML
+    private void handleUserInput() {
+        String input = userInput.getText();
+        if (input == null || input.isBlank()) {
+            return;
+        }
+
+        addUserLine(input);
+        String response = yoyo.respond(input);
+        addBotLine(response);
+
+        userInput.clear();
+
+        yoyo.shouldExit(input);// Close gracefully after a short delay (optional)
+// Platform.exit();
+    }
```

Comment:
It's excellent that you check for null or blank input before processing. This prevents unnecessary operations and potential errors down the chain. The method yoyo.shouldExit(input) is called, and there's a commented-out Platform.exit(). This looks like unfinished logic for handling an exit command (like "bye").
```suggestion
if (yoyo.shouldExit(input)) {
    // Add a farewell message before exiting?
    addBotLine("Goodbye! Hope to see you again soon!");
    // Exit after a short delay to show the message
    PauseTransition delay = new PauseTransition(Duration.seconds(1.5));
    delay.setOnFinished(event -> Platform.exit());
    delay.play();
} else {
    String response = yoyo.respond(input);
    addBotLine(response);
}
```